lib/utils/batcher/state.go

```diff
@@ -51,16 +56,22 @@ type StateConfig struct {
 	// OnExitOverloaded is called when transitioning from Overloaded to Normal state.
 	// This callback is invoked when batch size drops below threshold/2.
 	//
-	// WARN: For simplicity there is not timer-based transition back to Normal state,
-	// it only happens when batch size drops below threshold/2.
-	// So if the last batch is larger and there are no new events, the state will remain Overloaded.
 	// Is not provided, no action is taken on state transition.
 	OnExitOverloaded func()
 
 	// Threshold is the batch size threshold for state transitions.
 	// Normal -> Overloaded: when batchSize >= Threshold
 	// Overloaded -> Normal: when batchSize < Threshold/2
 	Threshold int
+
+	// OverloadedIdle is the quiet-period after which the monitor will (if still in
+	// StateOverloaded) automatically return to StateNormal even if no new batches arrive.
+	// If zero, the timer-based transition is disabled and the state will only return
+	// to Normal via an explicit size drop (< Threshold/2).
+	OverloadedIdle time.Duration
```

Comment:
Nit: I'd name this explicitly "timeout".
```suggestion
	OverloadedIdleTimeout time.Duration
```